### PR TITLE
Pass keyword arguments to the success result block

### LINKED ISSUE #123: [REQUEST] `run` should provide kwargs
Currently, you can use `run` to invoke an operation in a controller. If it's successful, the passed block will be invoked and receives the `result` or `ctx` object from the run operation. http://trailblazer.to/2.1/docs/trailblazer.html#trailblazer-rails-run

In endpoint, you can do something like below, which is really convenient: when the OP was run successfully, its result/ctx is passed to the block, plus the extractable keyword arguments are passed to the block.

```ruby
run Song::Operation::Create do |ctx, model:, **|
```

Using keyword arguments also provides a stronger interface to the rendering.

### PR DESCRIPTION
**Pull request by pedroaugustofsilva**
I can't run the tests and I don't know if it's necessary.

But if it is, I would like to say that I tried to install the gems and run the rake test, but I received `test/dummy/config/application.rb:5:in 'require': cannot load such file -- active_record/railtie (LoadError)`.

I could create a contribution guide to this project, to make it easier for new contributors.

resolves #123

### CODE REVIEW

**Comment by apotonick** [comment]
Hey @pedroaugustofsilva! Thanks! A few test cases covering this behavior is indispensable. I am trying to fix the test suite as I type, not sure who broke this *cough @seuros :wink:

**Comment by richardboehme** [comment]
Hey! When running tests you have to bundle one of the rails-Gemfiles found in the gemfiles folder. You can either use Appraisal to install those gems or select one Gemfile manually using:

´´´
BUNDLE_GEMFILE=gemfiles/rails_6.1.gemfile bundle install
BUNDLE_GEMFILE=gemfiles/rails_6.1.gemfile rake db:create db:schema:load test
```

**Comment by seuros** [comment]
@pedroaugustofsilva  check #125

**Comment by pedroaugustofsilva** [comment]
> Hey! When running tests you have to bundle one of the rails-Gemfiles found in the gemfiles folder. You can either use Appraisal to install those gems or select one Gemfile manually using:
> 
> ```
> BUNDLE_GEMFILE=gemfiles/rails_6.1.gemfile bundle install
> BUNDLE_GEMFILE=gemfiles/rails_6.1.gemfile rake db:create db:schema:load test
> ```

I tried to run with the gemfiles, but it wasn't working. I'll try using `appraisal`.

Thanks!

**Comment by apotonick** [comment]
@pedroaugustofsilva You can also hit us up directly here https://trailblazer.zulipchat.com

File: test/dummy/app/controllers/extractable_keywords_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class ExtractableKeywordsController < ApplicationController
+  def extract_one
+    run ExtractableKeywords::Operation::Extract do |_ctx, keyword_1:, **|
+      render html: %{<h1>#{keyword_1}</h1>}.html_safe
```
Comment (seuros) [inline test/dummy/app/controllers/extractable_keywords_controller.rb:4]: "render plain:"

File: lib/trailblazer/rails/controller.rb [hunk 2]
```diff
@@ -10,9 +10,15 @@ def run_v21(operation, *dependencies, **variables)
 
       _assign_trb_ivars(result)
 
-      yield(result) if result.success? && block_given?
-
       @_result = result
+
+      return unless result.success? && block_given?
```
Comment (apotonick) [inline lib/trailblazer/rails/controller.rb:15]: I'm not sure this tests what you intend. Maybe this should be `return if result.failure?`, only? We don't care about the block, anyway, in this case.

Comment (apotonick) [inline lib/trailblazer/rails/controller.rb:15]: This should be `return if result.success?` as we're only interested in this case, right?

File: test/integration/extractable_keywords_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (apotonick) [inline test/integration/extractable_keywords_controller_test.rb:1]: Great! :green_heart:

Comment (pedroaugustofsilva) [reply test/integration/extractable_keywords_controller_test.rb:1]: Thanks!

**Review by apotonick — APPROVED** [review]

**Comment by apotonick** [comment]
@yogeshjain999 to me this looks like great work and ready-to-be-shipped! :shipit:

**Review by seuros — APPROVED** [review]

**Review by yogeshjain999 — APPROVED** [review]
Yeah, looking good!!

**Comment by yogeshjain999** [comment]
Released in `2.4.1`, thanks @pedroaugustofsilva 🍻

**Pull request merged**